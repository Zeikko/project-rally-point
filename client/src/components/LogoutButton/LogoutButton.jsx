import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { logoutAction } from '../../actions/user-actions'
import Button from '../Button/Button'

class LogoutButton extends Component {
  constructor() {
    super()
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick() {
    const { dispatch } = this.props
    dispatch(logoutAction())
  }

  render() {
    return (
      <Button onClick={this.handleClick}>
        Logout
      </Button>
    )
  }
}

LogoutButton.propTypes = {
  dispatch: PropTypes.func.isRequired,
}

export default LogoutButton
