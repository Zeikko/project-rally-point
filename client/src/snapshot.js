import renderer from 'react-test-renderer'
import 'jest-styled-components'

export function matchComponentToSnapshot(component) {
  const json = renderer.create(component).toJSON()
  expect(json).toMatchSnapshot()
}