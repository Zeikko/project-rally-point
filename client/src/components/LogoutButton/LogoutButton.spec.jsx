import React from 'react'
import { shallow } from 'enzyme'
import { matchComponentToSnapshot } from '../../test/snapshot'
import LogoutButton from './LogoutButton'

describe('LogoutButton', () => {
  it('renders correctly', () => {
    matchComponentToSnapshot(<LogoutButton dispatch={() => {}} />)
  })

  it('logs user out', () => {
    const dispatch = jest.fn()
    const wrapper = shallow(<LogoutButton dispatch={dispatch} />)
    wrapper.find('button').simulate('click')
    expect(dispatch.mock.calls[0]).toEqual([
      {
        type: 'LOGOUT_REQUEST',
      },
    ])
  })
})
