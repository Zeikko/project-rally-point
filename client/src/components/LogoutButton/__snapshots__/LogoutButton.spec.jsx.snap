// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutButton renders correctly 1`] = `
<button
  href="/api/auth/login"
  onClick={[Function]}
>
  Logout
</button>
`;
